src/main/java/pl/javastart/restoffers/category/CategoryService.java

```diff
@@ -0,0 +1,46 @@
+package pl.javastart.restoffers.category;
+
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+@Service
+public class CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryDtoMapper categoryDtoMapper;
+
+    public CategoryService(CategoryRepository categoryRepository, CategoryDtoMapper categoryDtoMapper) {
+        this.categoryRepository = categoryRepository;
+        this.categoryDtoMapper = categoryDtoMapper;
+    }
+
+    Optional<CategoryDto> getCategoryById(Long id) {
+        return categoryRepository.findById(id)
+                .map(categoryDtoMapper::map);
+    }
+
+    List<CategoryDto> getAllCategories() {
+        return StreamSupport.stream(categoryRepository.findAll().spliterator(), false)
+                .map(categoryDtoMapper::map)
+                .toList();
```

Comment:
Jeśli  użyjesz `JpaRepository` zamiast `CrudRepository`, to findAll będzie zwracać List, a nie Iterator, więc będzie można użyć bardziej typowej składni:

```java
    List<CategoryDto> getAllCategories() {
        return categoryRepository.findAll().stream()
                .map(categoryDtoMapper::map)
                .toList();
    }
```